picard/file.py

```diff
@@ -284,6 +284,14 @@ def _make_filename(self, filename, metadata, settings=config.setting):
 
         if settings["rename_files"]:
             new_filename, ext = os.path.splitext(new_filename)
+            try:
+                # In case the filename is blank and only has the extension
+                # the real extension is in new_filename and ext is blank
+                if ext == '' and new_filename.lower() in map(unicode, self.EXTENSIONS):
+                    ext = new_filename
+                    new_filename = ''
+            except AttributeError:
```

Comment:
`register_formats` and `supported_formats` already don't work if `EXTENSIONS` is not a defined attribute.